### ISSUE TITLE
wxCaret with standard shape

### ISSUE THREAD

**Issue by AliKet** [wxWidgets/wxWidgets#22254]
As you know all, the caret may be represented using:
1- a thin vertical line.
2- a thick rectangle (character-sized) as the current implementation does.
3- the underscore shape (more common under **MSW**)
4- same as **1** but with indicator for language direction

IMHO we can easily add support for **1** and **3** directly to `wxCaret`
without asking for an external bitmap which will make using `wxCaret` a little bit
easier and more importantly add support to default caret shapes, that is:
- Make shape-2 the default under *nix
- Make shape-3 the default under Windows

I have a working implementation under both GTK3 (with little modification to [this](https://github.com/wxWidgets/wxWidgets/pull/22079))
and MSW.

Some screenshots showing 1 an 3 under Ubuntu 18.04:

Shape 1:
![Screenshot from 2022-04-01 15-45-13](https://user-images.githubusercontent.com/7704771/161287674-457cc808-1640-45bc-a1d4-3e71f5e184c8.png)

Shape 3:
![Screenshot from 2022-04-01 15-40-26](https://user-images.githubusercontent.com/7704771/161287817-365232a9-ad33-4948-9046-84bbb2705424.png)

What do you think ?

TIA!

**Comment by vadz** [comment]
I agree that it would be nice to support the standard caret shapes, but the question is, as usual, how much complexity/extra code will this add? Under MSW it should be relatively simple to support this, but I don't know about other platforms.

BTW, I have no idea if the size of the caret is interpreted correctly in high DPI, would you know this by chance? It would be nice if we could avoid specifying the caret size at all and just use the font size or even simply the window font to determine it automatically.
